src/garth/data/badge.py

```
@@ -0,0 +1,190 @@
+        Args:
+            badge_id: The Garmin badge ID
+            client: Optional HTTP client (uses default if not provided)
+
+        Returns:
+            Badge instance with full details
+        """
+        client = client or http.client
+        path = f"/badge-service/badge/detail/v2/{badge_id}"
+        data = client.connectapi(path)
+        if not data:
+            raise ValueError(f"No data returned from {path}")
```

Ruff (0.15.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)